lib/src/rules/avoid_dynamic_calls.dart

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import '../analyzer.dart';
+
+const _desc = r'Avoid method calls and property access on a "dynamic" target.';
+
+const _details = r'''
+
+**DO** avoid method calls or accessing properties on an object that is either
+explicitly or implicitly statically typed "dynamic". Dynamic calls are treated
+slightly different in every runtime environment and compiler, but most
+production modes (and even some development modes) have both compile size and
+runtime performance penalties associated with dynamic calls.
+
+Additionally, targets typed "dynamic" disables most static analysis, meaning it
+is easier to lead to a runtime "NoSuchMethodError" or "NullError" than properly
+statically typed Dart code.
+
+Note, that despite "Function" being a type, the semantics are close to identical
+to "dynamic", and calls to an object that is typed "Function" will also trigger
+this lint.
+
```

Comment:
Super awesome docs.

Would that they were all this good!

💎

Reply:
Thanks @pq!